io_scene_godot/converters/mesh.py

```diff
@@ -219,156 +102,249 @@ def validate_morph_mesh_modifiers(mesh_object):
+    if does not have one, return None"""
+    for modifier in mesh_object.modifiers:
+        if isinstance(modifier, bpy.types.ArmatureModifier):
+            return modifier.object.data
+    return None
+
+
+class MeshResourceExporter:
+    """Export a mesh resource from a blender mesh object"""
+    def __init__(self, mesh_object):
+        # blender mesh object
+        self.object = mesh_object
+
+        self.mesh_resource = None
+        self.has_tangents = False
+        self.vgroup_to_bone_mapping = dict()
+        self.mat_to_surf_mapping = dict()
+
+    def init_mesh_bones_data(self, skeleton_node):
+        """Find the mapping relation between vertex groups
+        and bone id"""
+        for bone_name, bone_id in skeleton_node.bone_name_to_id_map.items():
+            group = self.object.vertex_groups.get(bone_name)
+            if group is not None:
+                self.vgroup_to_bone_mapping[group.index] = bone_id
+
+    def export_mesh(self, escn_file, export_settings):
+        """Saves a mesh into the escn file """
+        # Check if it exists so we don't bother to export it twice
+        mesh = self.object.data
+        mesh_id = escn_file.get_internal_resource(mesh)
+
+        if mesh_id is not None:
+            return mesh_id
+
+        self.mesh_resource = InternalResource('ArrayMesh')
+
+        try:
+            self.make_arrays(
+                escn_file,
+                export_settings,
             )
-    for index, shape_key in extract_shape_keys(node.data.shape_keys):
+        except ValidationError as exception:
+            exception.args += (mesh.name, )
+            raise
+
+        mesh_id = escn_file.add_internal_resource(self.mesh_resource, mesh)
+        assert mesh_id is not None
+
+        return mesh_id
+
+    def make_arrays(self, escn_file, export_settings):
+        """Generates arrays of positions, normals etc"""
+        armature_data = get_modifier_armature_data(self.object)
+        if armature_data is not None:
+            original_pose_position = armature_data.pose_position
+            armature_data.pose_position = 'REST'
+            bpy.context.scene.update()
 
-        node.show_only_shape_key = True
-        node.active_shape_key_index = index
-        shape_key.value = 1.0
+        if not export_settings['use_mesh_modifiers']:
+            for modifier in self.object.modifiers:
+                if not isinstance(modifier, bpy.types.ArmatureModifier):
+                    modifier.show_render = False
 
-        mesh = node.to_mesh(bpy.context.scene,
-                            True,
-                            "RENDER")
+        mesh = self.object.to_mesh(bpy.context.scene,
+                                   True,
+                                   "RENDER")
+
+        # Prepare the mesh for export
         triangulate_mesh(mesh)
 
-        if has_tangents:
+        uv_layer_count = len(mesh.uv_textures)
+        if uv_layer_count > 2:
```

Comment:
I think this was copied over from before, but using the `min` function would be prettier